pyuvdata/tests/test_uvflag.py

```diff
@@ -169,18 +169,13 @@ def test_init_waterfall_flag():
 def test_init_waterfall_copy_flags():
     uv = UVCal()
     uv.read_calfits(test_c_file)
-    uvf = uvtest.checkWarnings(UVFlag, [uv], {'copy_flags': True, 'mode': 'flag', 'waterfall': True},
-                               nwarnings=1, message='Copying flags into waterfall')
-    assert not hasattr(uvf, 'flag_array')  # Should be metric due to copy flags
-    assert uvf.metric_array.shape == (uv.Ntimes, uv.Nfreqs, uv.Njones)
-    assert uvf.weights_array.shape == (uv.Ntimes, uv.Nfreqs, uv.Njones)
-    assert uvf.type == 'waterfall'
-    assert uvf.mode == 'metric'
-    assert np.all(uvf.time_array == np.unique(uv.time_array))
-    assert np.all(uvf.freq_array == uv.freq_array[0])
-    assert np.all(uvf.polarization_array == uv.jones_array)
-    assert 'Flag object with type "waterfall"' in uvf.history
-    assert pyuvdata_version_str in uvf.history
+    pytest.raises(NotImplementedError, UVFlag, uv, copy_flags=True, mode='flag',
+                  waterfall=True)
+
+
+def test_init_invalid_input():
+    # input is not UVData, UVCal, path, or list/tuple
+    pytest.raises(ValueError, UVFlag, 14)
```

Comment:
I think this line tests the following code https://github.com/RadioAstronomySoftwareGroup/pyuvdata/blob/e1e023c182a0ee6a38490f372c98a44ffdcfacb6/pyuvdata/uvflag.py#L137-L139
it would be helpful if you could catch the exception and make sure you are raising the right one. See example below
https://github.com/RadioAstronomySoftwareGroup/pyuvdata/blob/680664d11c73f4afed6837c79aa7e4ff906242b4/pyuvdata/tests/test_ms.py#L94-L96

Reply:
neato